src/contracts/Test/IbexaTestKernel.php

```diff
@@ -0,0 +1,184 @@
+    {
+        $container->loadFromExtension('doctrine', [
+            'dbal' => [
+                'memory' => true,
+                'driver' => 'pdo_sqlite',
+                'platform_service' => SqliteDbPlatform::class,
+                'logging' => false,
+            ],
+        ]);
+
+        $definition = new Definition(Repository\Tests\LegacySchemaImporter::class);
+        $definition->setPublic(true);
+        $definition->setArgument(0, new Reference('doctrine.dbal.default_connection'));
+        $container->setDefinition('test.ibexa.migrations.schema_importer', $definition);
+
+        $definition = new Definition(FixtureImporter::class);
+        $definition->setPublic(true);
+        $definition->setArgument(0, new Reference('doctrine.dbal.default_connection'));
+        $container->setDefinition('test.ibexa.migrations.fixture_importer', $definition);
+
+        $definition = new Definition(SqliteDbPlatform::class);
+        $definition->addMethodCall('setEventManager', [
+            new Reference('doctrine.dbal.default_connection.event_manager'),
+        ]);
+        $container->setDefinition(SqliteDbPlatform::class, $definition);
+    }
+
+    private static function prepareEzPlatformFramework(ContainerBuilder $container): void
+    {
+        $container->setParameter('io_root_dir', '');
+        $container->setParameter('kernel.secret', 'foobar');
+        $container->loadFromExtension('ezplatform', [
+            'siteaccess' => [
+                'default_siteaccess' => '__default_site_access__',
+                'list' => [
+                    'default' => '__default_site_access__',
+                    'second' => '__second_site_access__',
+                    'ger' => 'ger',
+                    'eng' => 'eng',
+                    'ku6"H' => 'ku6"H',
+                ],
+                'match' => null,
+            ],
+            'repositories' => [
+                'default' => [
+                    'storage' => null,
+                    'search' => [
+                        'engine' => 'legacy',
+                        'connection' => 'default',
+                    ],
+                ],
+            ],
+        ]);
+
+        $container->loadFromExtension('security', [
+            'providers' => [
+                'default' => [
+                    'id' => 'ezpublish.security.user_provider',
+                ],
+            ],
+            'firewalls' => [
+                'main' => [
+                    'anonymous' => null,
+                ],
+            ],
+        ]);
+
+        $container->loadFromExtension('framework', [
+            'test' => true,
+            'session' => [
+                'storage_id' => 'session.storage.mock_file',
+            ],
+            'cache' => [
+                'app' => 'cache.adapter.array',
+            ],
+            'router' => [
+                'resource' => 'foo',
+            ],
+        ]);
```

Comment:
It's best that we load this configuration from Yaml files. It would allow to optimize the spaghetti code here.

Reply:
Replaced with YAML files.